src/test/java/space/obminyashka/items_exchange/service/AuthServiceTest.java

```diff
@@ -1,74 +1,141 @@
+                () -> verify(jwtTokenService).getRefreshTokenExpiration(any(ZonedDateTime.class))
+        );
+    }
+
+    private UserLoginResponseDto createTestDto() {
+        var dto = new UserLoginResponseDto();
+        dto.setEmail("user@mail.ua");
+        dto.setUsername(EXPECTED_USERNAME);
+        dto.setRole(new Role(UUID.randomUUID(), "ROLE_USER", List.of()));
+        return dto;
+    }
+
+    private void mockJwtTokenServiceForCreationAccessAndRefreshTokens() {
+        when(jwtTokenService.createAccessToken(anyString(), any(Role.class))).thenReturn(JWT_TOKEN);
+        when(jwtTokenService.getAccessTokenExpiration(anyString())).thenReturn(LocalDateTime.MAX);
+        when(jwtTokenService.getRefreshTokenExpiration(any(ZonedDateTime.class))).thenReturn(REFRESH_TOKEN);
+    }
+
+    private void checkAccessAndRefreshTokenFieldsInUserLoginResponseDto(UserLoginResponseDto dto) {
+        assertThat(dto)
+                .hasFieldOrPropertyWithValue("accessToken", JWT_TOKEN)
+                .hasFieldOrPropertyWithValue("refreshToken", REFRESH_TOKEN)
                 .hasFieldOrProperty("accessTokenExpirationDate")
                 .hasFieldOrProperty("refreshTokenExpirationDate");
-        verify(jwtTokenService).createAccessToken(usernameCaptor.capture(), any(Role.class));
-        assertEquals("user", usernameCaptor.getValue());
     }
 
-    private void mockJwtTokenData() {
-        when(jwtTokenService.createAccessToken(anyString(), any())).thenReturn(jwtToken);
-        when(jwtTokenService.getAccessTokenExpiration(jwtToken)).thenReturn(LocalDateTime.MAX);
+    @Test
+    void logout_whenPositiveFlow_shouldSuccessfullyLogout() {
+        authService.logout(JWT_TOKEN, EXPECTED_USERNAME);
+
+        assertAll(
+                () -> verify(jwtTokenService).invalidateAccessToken(tokenCaptor.capture()),
+                () -> assertEquals(JWT_TOKEN, tokenCaptor.getValue()),
+                () -> verify(refreshTokenService).deleteByUsername(usernameCaptor.capture()),
+                () -> assertEquals(EXPECTED_USERNAME, usernameCaptor.getValue())
+        );
+    }
+
+    @Test
+    void renewAccessTokenByRefresh_whenRefreshTokenNotEmpty_shouldCreateNewAccessToken() throws RefreshTokenException {
+        when(refreshTokenService.renewAccessTokenByRefresh(any())).thenReturn(Optional.of(JWT_TOKEN));
+
+        var refreshTokenDto = authService.renewAccessTokenByRefresh(REFRESH_TOKEN);
+
+        assertAll(
+                () -> checkAccessAndRefreshTokenFieldsInRefreshTokenResponseDto(refreshTokenDto),
+                () -> verify(refreshTokenService).renewAccessTokenByRefresh(REFRESH_TOKEN),
+                () -> verify(jwtTokenService).getAccessTokenExpiration(JWT_TOKEN),
+                () -> verify(jwtTokenService).getRefreshTokenExpiration(any(ZonedDateTime.class))
+        );
+    }
+
+    private void checkAccessAndRefreshTokenFieldsInRefreshTokenResponseDto(RefreshTokenResponseDto dto) {
```

Comment:
Why don't you get a generic object here and simply combine that method with **checkAccessAndRefreshTokenFieldsInUserLoginResponseDto** ?